sdk/core/Azure.Core/tests/HttpMessageTests.cs

```diff
@@ -241,6 +241,72 @@ public void AppliesHandlerBeforeResponseClassifier()
             Assert.IsTrue(message.ResponseClassifier.IsErrorResponse(message));
         }
 
+        [Test]
+        public void AppliesResponseClassifier()
+        {
+            HttpMessage message = new HttpMessage(new MockRequest(), default);
+            message.ApplyRequestContext(new RequestContext(), HeadResponseClassifier.Instance);
+
+            message.Response = new MockResponse(204);
+            Assert.IsFalse(message.ResponseClassifier.IsErrorResponse(message));
+
+            message.Response = new MockResponse(304);
+            Assert.IsTrue(message.ResponseClassifier.IsErrorResponse(message));
+
+            message.Response = new MockResponse(404);
+            Assert.IsFalse(message.ResponseClassifier.IsErrorResponse(message));
+
+            message.Response = new MockResponse(500);
+            Assert.IsTrue(message.ResponseClassifier.IsErrorResponse(message));
+        }
+
+        [Test]
+        public void ChainsClassifiers_StatusCodes()
+        {
+            RequestContext context = new RequestContext();
+            context.AddClassifier(404, true);
+            context.AddClassifier(500, false);
+
+            HttpMessage message = new HttpMessage(new MockRequest(), default);
+            message.ApplyRequestContext(context, HeadResponseClassifier.Instance);
+
+            message.Response = new MockResponse(204);
+            Assert.IsFalse(message.ResponseClassifier.IsErrorResponse(message));
+
+            message.Response = new MockResponse(304);
+            Assert.IsTrue(message.ResponseClassifier.IsErrorResponse(message));
+
+            message.Response = new MockResponse(404);
+            Assert.IsTrue(message.ResponseClassifier.IsErrorResponse(message));
+
+            message.Response = new MockResponse(500);
+            Assert.IsFalse(message.ResponseClassifier.IsErrorResponse(message));
+        }
+
+        [Test]
+        public void ChainsClassifiers_StatusCodesAndHandlers()
+        {
+            RequestContext context = new RequestContext();
+            context.AddClassifier(404, true);
+            context.AddClassifier(500, false);
+            context.AddClassifier(new StatusCodeHandler(404, false));
```

Comment:
nit: I think this should differ from the default classification that HeadResponseClassifier would have returned just to ensure it is taking precedence.

Reply:
Yeah, good call.